service/app/controllers/otp/validator/OTPRequestValidator.java

```diff
@@ -0,0 +1,49 @@
+package controllers.otp.validator;
+
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.util.EmailValidator;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.ProjectUtil;
+import org.sunbird.common.request.BaseRequestValidator;
+import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
+
+public class OTPRequestValidator extends BaseRequestValidator {
+  private static final int ERROR_CODE = ResponseCode.CLIENT_ERROR.getResponseCode();
+
+  public void validateGenerateOTPRequest(Request otpRequest) {
+    validateParam(
+        (String) otpRequest.getRequest().get(JsonKey.KEY),
+        ResponseCode.mandatoryParamsMissing,
+        JsonKey.KEY);
+    validateParam(
+        (String) otpRequest.getRequest().get(JsonKey.TYPE),
+        ResponseCode.mandatoryParamsMissing,
+        JsonKey.TYPE);
+
+    validateKeyFormat(otpRequest);
+  }
+
+  private void validateKeyFormat(Request otpRequest) {
+    if (JsonKey.EMAIL.equalsIgnoreCase((String) otpRequest.getRequest().get(JsonKey.TYPE))) {
+      if (!EmailValidator.isEmailValid((String) otpRequest.getRequest().get(JsonKey.KEY))) {

```

Comment:
Add validateEmail and validatePhone to BaseRequestValidator so it can be reused in verify OTP request as well.